tsig.go

```diff
@@ -74,6 +74,24 @@ func (key tsigHMACProvider) Verify(msg []byte, t *TSIG) error {
 	return nil
 }
 
+type tsigSecretProvider map[string]string
```

Comment:
This is good for a default TsigProvider, however it still doesn't solve the problem described in 

https://github.com/miekg/dns/issues/1325#issuecomment-1018712904

We need some way for Generate to be able to unwrap the Message so that we can see the query name. This is specifically useful in situations where multiple tsigs have the same name, but exist under a different zone.

Reply:
That can be handled either in a separate PR or externally, but yes we do need to do *something*.

Reply:
How common a use case is that do you think?

Reply:
I would say it's fairly common. I ran into this problem early on with this library when trying to serve multiple zones that had different primaries, therefore different TSIGs on each primary meaning they could not use the TSIG name as the zone Name.

Reply:
Hey @tmthrgd have you thought any more about how we can best obtain the query name from the Generate function. I think having the opposite of TsigBuffer available in a function would make this possible. Alternatively we could consider passing in a context? It would be useful to have contexts everywhere anyway so that we could pass data around without needing to alter signatures.